ls1intum/artemis-android
# Add Profile pictures in remaining places

### Problem Description
In the previous PRs #111 and #163 Profile pictures has only been added to the main chat lists. But they are not displayed in other contexts.

### Changes
 
- Added profile picture in ...
  - Conversation overview
  - Conversation header for DMs
  - User search (channel creation)
  - Conversation members
  - User settings screen

- Click on profile picture shows popup dialog
  - Disabled for settings and user search
  - The "send message" button has been postponed due to ongoing API changes, see #213 

- Misc
  - Changed channel icons, so that only one icon get displayed per channel. Previously a single channel could have the icons for channel (hash), announcement, and archived. Now only the must important icon is displayed, similar to iOS.


### Steps for testing
For the following steps, you can confirm each time whether clicking on the profile picture shows the user dialog or not
- Go to the app settings
  - See that the profile picture is displayed here
- Go back to conversation overview
  - For eg announcement channels (or archived ones), only a single icon is displayed
  - See that in the "Direct messages" section the profile pictures are displayed
- Click on a non-DM chat
  - See that the channel icon is displayed in the title
  - Click on settings
  - See that the profile pictures are displayed in the members section 
- Navigate back and click on a DM chat
  - See the profile picture in the title
- Navigate back and click the create channel floating action button
  - Search for a user
  - See that the profile picture is displayed 


### Screenshots
Settings
![grafik](https://github.com/user-attachments/assets/a8740bb3-d0cf-4d4b-bafa-25ccb1e21d5d)

Conversation Overview
![grafik](https://github.com/user-attachments/assets/0225c976-9029-41e4-a43e-e744e2493d57)

Conversation Header
![grafik](https://github.com/user-attachments/assets/ead3e37d-8b25-40d2-bcc3-f591d02fa4a4)

Members
![grafik](https://github.com/user-attachments/assets/34175e0c-e736-4e83-bc55-1e423708f08e)

Popup Dialog
![grafik](https://github.com/user-attachments/assets/11c8f791-85b7-49d9-94b4-a9f33c5fc8df)
